client/src/components/DetailView.js

```diff
@@ -58,13 +58,15 @@ export default class DetailView extends Component {
 
   render() {
 
-    let boro, block, lot, ownernames, userOwnernames;
+    let boro, block, lot, ownernames, userOwnernames, takeActionURL;
     if(this.props.addr) {
       ({ boro, block, lot } = Helpers.splitBBL(this.props.addr.bbl));
+      takeActionURL = 'https://app.justfix.nyc/ddo?address=' + encodeURIComponent((this.props.addr.housenumber + ' ' + this.props.addr.streetname).trim()) + '&borough=' + encodeURIComponent(this.props.addr.boro) + '/?utm_source=whoownswhat_take_action';
```

Comment:
So the borough is supposed to be one of the borough constants defined here:

https://github.com/JustFixNYC/tenants2/blob/master/common-data/borough-choices.json

That basically means that you will want to uppercase the borough name and convert spaces to underscores...

Reply:
Err also see the comment I left here: https://github.com/JustFixNYC/who-owns-what/pull/175/files#r314397915

Shouldn't both of these almost-identical lines of code in `DetailView.js` and `NychaPage.js` really be factored out into a separate function, e.g. `createTakeActionURL()`?